packages/core/admin/ee/server/services/review-workflows/stages.js

```diff
@@ -77,20 +85,13 @@ module.exports = ({ strapi }) => {
 
       // Update stages and assign entity stages
       return strapi.db.transaction(async ({ trx }) => {
-        // Create the new stages
-        const createdStages = await this.createMany(created, { fields: ['id'] });
-        // Put all the newly created stages ids
-        const createdStagesIds = map('id', createdStages);
-
-        // Update the workflow stages
-        await mapAsync(updated, (stage) => this.update(stage.id, stage));
-
+        // ⚠️ Delete first as we can have a limit on stages
         // Delete the stages that are not in the new stages list
         await mapAsync(deleted, async (stage) => {
           // Find the nearest stage in the workflow and newly created stages
           // that is not deleted, prioritizing the previous stages
           const nearestStage = findNearestMatchingStage(
-            [...srcStages, ...createdStages],
+            srcStages,
```

Comment:
I think we should keep the original order here. the source stages also need the created stages at the end. If you delete all stages and add a new one, the computed `nearestStage` will be undefined, and entities stage will be deleted.

Reply:
We can't have the created stages here as the creation is done afterward.

Reply:
but we should because of that point

Reply:
Oh but as I removed the validation, I may not need to reorder things here too. Will change this